### PR TITLE
Feature annotation layer

### PR DESCRIPTION
**Pull request by kotfic**

### CODE REVIEW

**Comment by kotfic** [comment]
Note: This will resolve https://github.com/OpenGeoscience/geonotebook/issues/31

**Comment by kotfic** [comment]
@danlamanna I still need to write some tests and fix up some documentation,  but if you could start reviewing that would be a big help.  Thanks!

File: geonotebook/kernel.py [hunk 1]
```diff
@@ -404,8 +407,16 @@ def handle_comm_open(self, comm, msg):
...
         # THis should be handled in a callback that is fired off
         # When set protocol etc is complete.
-        self.geonotebook.add_layer(None, name="osm_base", layer_type="osm",
-                                   vis_url="http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png")
+        self.geonotebook.add_layer(
+            None, name="osm_base", layer_type="osm",
+            vis_url="http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
+            system_layer=True)
+
+        self.geonotebook.add_layer(
```
Comment (danlamanna) [inline geonotebook/kernel.py]: I can't see any immediate problems with this, but isn't there a bit of a code smell around the fact that an annotation layer exists even when there are no annotations?

The alternative being to create an annotation layer when the first annotation is drawn, I'm not sure if that's cleaner.

File: geonotebook/layers.py [hunk 2]
```diff
@@ -151,92 +217,86 @@ def next(self):
...
 
-# GeonotebookStack supports dict-like indexing on a list
-# of Geonotebook Layers. We could implement this with an
-# OrderedDict,  but i think we are eventually going to want
-# to support re-ordering,  potentially serializing etc so it
-# Seems like putting it in its own class is best for now.
 
-# TODO: support slices other list functionality etc
-# TODO: convert to collections.mutablesequence
 class GeonotebookStack(object):
```
Comment (danlamanna) [inline geonotebook/layers.py]: This wasn't added in this PR - but is this name appropriate? It's not technically a stack, and it's designed specifically to store Layer objects.

**Comment by danlamanna** [comment]
The example notebooks are really well done, though I think 3 and 4 could use more docs (not necessarily on this PR).

File: geonotebook/kernel.py (hunk 1 above)
Comment (kotfic) [inline geonotebook/kernel.py]: The annotation layer exists to create the potential for annotations.  Even if there are no annotations there should always be the potential for annotations.

**Comment by kotfic** [comment]
@danlamanna PTAL

**Comment by danlamanna** [comment]
Could you address my comment about matplotlib and pyproj? Only asking because if I install in a clean virtualenv and open the first example the console throws a "no module named matplotlib" exception immediately.

Otherwise LGTM.

**Pull request merged**